### DISCUSSION TITLE
RDF-Delta with Fuseki UI

### DISCUSSION THREAD

**Discussion by tomkxy** [afs/rdf-delta#122] [Q&A] [answered]
Would it be possible to package RDF-Delta with the Fuseki UI? And if yes, what would be required to make it running?

**Comment by afs** [comment]
Also: issue #17.  The Discussion area is better for this.

The version of Fuseki in RDF Delta is the "main" one, which is not webapp based. The UI comes with the webapp packaging (war file or standalone server running a webapp).

The core Fuseki engine is oblivious to RDF Delta; the synchronization is on the dataset.

Which aspects of the UI are you using? Some parts, like the query interface and data update are detachable.

The admin - create datasets - is more complicated for RDF Delta. Teh Fuseki server configuration isn't replicated in RDF Delta and it would need to be coordinated with setting up the patch server for a new dataset and have Fuseki server react to system changes, not just dataset changes.

One more point - use I know of to date has been using automation of configuration through rather than having a manual-like admin UI/API to manage the datasets.

Adding query UI would be relatively easy; RDF Delta does not need to be involved. It is a matter of adding it to Fuseki main. Fuseki main already supports serving static files. RDF Delta does not need to be involved.

So if it is for the query and data update functionality pages could be added to Fuseki main and woudl work here in the RDF Delta packaging of Fuseki.

See also: https://issues.apache.org/jira/browse/JENA-2087

**Comment by tomkxy** [comment, marked as answer]
Thanks for the fast reply. Indeed, I am only looking at the query interface and update part. Although, I understand that this is available through the command line, the GUIs is too many people more convenient.

**Comment by afs** [comment]
Yes, everyone loves a GUI! Fuseki will server the file using `--base` as the static file area and YASGUI can be used [with a very basic webpage](https://gist.github.com/afs/a87790dc6414cef1bc86ae48acc20bf1).

`fuseki --base DIRECTORY --mem /ds`

It would be nice to have query, update and file upload when using Fuseki for basic 

The Fuseki/webapp code is quite old - this would be a good point to rewrite using (insert JS framework de jour).

The other approach is to identify an existing general purpose SPARQL browser app. What is the current state of open source SPARQL UI's?

**Comment by afs** [comment]
By the way: 

There are Jupyter notebooks for SPARQL endpoints as well as a [SPARQL plugin for Apache Zeppelin](https://zeppelin.apache.org/docs/0.10.0/interpreter/sparql.html).

[Bob DuCharme blog](https://github.com/paulovn/sparql-kernel).

[Jupyter kernel for SPARQL](https://github.com/paulovn/sparql-kernel).

There's an [AWS Neptune Jupyter Notebook for SPARQL / Gremlin/ Cypher](https://github.com/aws/graph-notebook).
